frontend/app/src/pages/home/join.tsx

```diff
@@ -1,11 +1,50 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { io } from "socket.io-client";
+
 import { Page } from "@/components";
+import { MATCHING_EVENTS } from "@/constants/matching";
+import { ROUTE } from "@/constants/route";
 import { FindingMatchCard, SelectPreferencesCard } from "@/features/matching";
+import { env } from "@/lib/env";
+import { Matching, Preferences, matchingSchema } from "@/types/matching";
 
 function JoinPage() {
+  const navigate = useNavigate();
+  const socket = io(env.VITE_MATCHING_SERVICE_URL);
```

Comment:
btw can add a `TODO:` on this because eventually we want the connection to be made thru the API gateway and not directly to the matching service URL